clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusV2ClusterCachingAgent.java

```diff
@@ -0,0 +1,628 @@
+    this.titusLoadBalancerClient = titusClientProvider.getTitusLoadBalancerClient(account, region.getName());
+    this.awsLookupUtil = awsLookupUtil;
+    this.pollIntervalMillis = pollIntervalMillis;
+    this.timeoutMillis = timeoutMillis;
+    this.registry = registry;
+
+    this.metricId = registry.createId("titus.cache.cluster").withTag("account", account.getName()).withTag("region", region.getName());
+  }
+
+  @Override
+  public long getPollIntervalMillis() {
+    return pollIntervalMillis;
+  }
+
+  @Override
+  public long getTimeoutMillis() {
+    return timeoutMillis;
+  }
+
+  @Override
+  public String getProviderName() {
+    return TitusCachingProvider.PROVIDER_NAME;
+  }
+
+  @Override
+  public String getAgentType() {
+    return account.getName() + "/" + region.getName() + "/" + TitusV2ClusterCachingAgent.class.getSimpleName();
+  }
+
+  @Override
+  public String getOnDemandAgentType() {
+    return getAgentType() + "-OnDemand";
+  }
+
+  @Override
+  public OnDemandMetricsSupport getMetricsSupport() {
+    return metricsSupport;
+  }
+
+  @Override
+  public Collection<AgentDataType> getProvidedDataTypes() {
+    return types;
+  }
+
+  @Override
+  public Optional<Map<String, String>> getCacheKeyPatterns() {
+    Map<String, String> cachekeyPatterns = new HashMap<>();
+    cachekeyPatterns.put(SERVER_GROUPS.ns, Keys.getServerGroupV2Key("*", "*", account.getName(), region.getName()));
+    return Optional.of(cachekeyPatterns);
+  }
+
+  @Override
+  public boolean handles(OnDemandType type, String cloudProvider) {
+    return type == OnDemandType.ServerGroup && cloudProvider.equals(titusCloudProvider.getId());
+  }
+
+  @Override
+  public OnDemandResult handle(ProviderCache providerCache, Map<String, ?> data) {
+    Long startTime = System.currentTimeMillis();
+
+    if (!data.containsKey("serverGroupName") || !data.containsKey("account") || !data.containsKey("region")) {
+      return null;
+    }
+
+    if (!account.getName().equals(data.get("account"))) {
+      return null;
+    }
+
+    if (!region.getName().equals(data.get("region"))) {
+      return null;
+    }
+
+    Job job = metricsSupport.readData( () -> {
+      try {
+        return titusClient.findJobByName(data.get("serverGroupName").toString());
+      } catch (io.grpc.StatusRuntimeException e) {
+        return null;
+      }
+    });
+
+    OnDemandResult onDemandResult = onDemand(providerCache, job, data);
+    PercentileTimer
+      .get(registry, metricId.withTag("operation", "handleOnDemand"))
+      .record(System.currentTimeMillis() - startTime, MILLISECONDS);
+
+    return onDemandResult;
+  }
+
+  /**
+   * Avoid writing cache results to both ON_DEMAND and SERVER_GROUPS, etc.
+   *
+   * By writing a minimal record to ON_DEMAND only, we eliminate significant overhead (redis and network) at the cost
+   * of an increase in time before a change becomes visible in the UI.
+   *
+   * A change will not be visible until a caching cycle has completed.
```

Comment:
if we aren't going to see a change until a caching cycle completes, and we aren't reading through any intermediate caches, is there value in onDemand here?  cc: @ajordens think you touched it last

Reply:
if we are down to 15s for a caching agent run with the new split changes, maybe we don't need onDemand complexity?

Reply:
I'll defer to @ajordens - this is just a cary over of his onDemand implementation.

Reply:
we still need it IMO